src/NuGet.Core/NuGet.Protocol/Providers/PackageSearchResourceV3Provider.cs

```diff
@@ -23,6 +25,16 @@ public PackageSearchResourceV3Provider()
             if (serviceIndex != null)
             {
                 var endpoints = serviceIndex.GetServiceEntryUris(ServiceTypes.SearchQueryService);
+
+                foreach (Uri endpoint in endpoints)
+                {
+                    PackageSource serviceSource = new PackageSource(endpoint.ToString(), endpoint.ToString());
+                    if (serviceSource.IsHttp && !serviceSource.IsHttps && !source.PackageSource.AllowInsecureConnections)
```

Comment:
Is it possible for isHttp and isHttps both ever evaluate to true? If not, this is also redudant.

Reply:
https is a subset of http. Http would be true for both https and non https